python_transport/setup.py

```diff
@@ -23,6 +23,30 @@
     long_description = f.read()
 
 
+def custom_scheme():
+    def custom_version(version):
+        tag = str(version.tag)
+        if version.exact:
+            return tag
+
+        # split on rc and compute a pre-release tag for the next patch
+        tag_components = tag.rsplit("rc")
+        major, minor, patch = tag_components[0].split(".")
+        patch = int(patch) + 1
+        distance = version.distance
```

Comment:
Are distance and patch number really different? It looks like both are the number of commits since last tag release? (except if latest tag has a patch value different that 0, right?)
So if we release 1.5.2, then any master commit will create 1.5.3.1, 1.5.4.2....

If the case, I wouldn't have touch the patch version. I don't really like to have official release 1.5.2 be followed by 1.5.15 for example.

What do you think?

Reply:
Lets say you have 1.5.2 tagged, the next commit will be 1.5.3.dev1 (first commit towards the next minor release). Basically the distance will be the amount of commits bundled in for the next release.

Reply:
Ok I think I get it now ;-)
So if latest tag is 1.5.2, then you will have 1.5.3.dev1, 1.5.3.dev2, and so on so next official could be 1.5.3!

I didn't read well the code and I was thinking that we would have ended up with 1.5.3.dev1, 1.5.4.dev2...
Sorry for that ;-) I didn't realize that patch is always from last tag and not incremented for each patch.

Then I'm fine!